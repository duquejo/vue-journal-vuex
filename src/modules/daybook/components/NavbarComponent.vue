<template>
  <nav class="navbar bg-primary">

    <a href="" class="navbar-brand text-white">
      <img src="assets/logo.png" alt="Vue logo" class="inline-block align-text-top mx-2" height="24"/>
      {{ username }}
    </a>

    <div class="d-flex">
      <button @click="onLogout" class="btn btn-outline-info mx-2">
        <i class="fa fa-sign-out-alt" />
      </button>
    </div>

  </nav>
</template>

<script>
import useAuth from '@/modules/auth/composables/useAuth';
import { useRouter } from 'vue-router';

export default {
  setup() {

    const router = useRouter();
    const { username, logoutUser } = useAuth();

    return {
      username,

      onLogout: () => {
        router.push({ name: 'login' });
        logoutUser();
      }
    };
  },
}
</script>