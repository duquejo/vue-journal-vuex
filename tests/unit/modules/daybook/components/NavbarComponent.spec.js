import { shallowMount } from "@vue/test-utils";
import NavbarComponent from '@/modules/daybook/components/NavbarComponent';
import createVuexStore from "../../../mocks/mock-store";
import { useRouter } from 'vue-router';

jest.mock('vue-router', () => ({
  useRoute: jest.fn(),
  useRouter: jest.fn(() => ({
    push: () => {}
  }))
}));

describe('NavbarComponent.vue', () => {

  const store = createVuexStore({
    user: {
      name: 'John Doe',
      email: 'foo@bar.com',
    },
    status: 'authenticated',
    idToken: 'abc-123',
    refreshToken: 'xyz-123',
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should show the component as needed', () => {
    const wrapper = shallowMount(NavbarComponent, {
      global: {
        plugins: [ store ],
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should logout when logout button is pressed and redirect', async () => {

    const push = jest.fn();
    useRouter.mockImplementationOnce(() => ({ push }));

    const wrapper = shallowMount(NavbarComponent, {
      global: {
        plugins: [ store ],
      }
    });

    await wrapper.find('button').trigger('click');

    expect(push).toHaveBeenCalledWith({ name: 'login' });
    expect(store.state.auth).toEqual({
      user: null,
      status: 'not-authenticated',
      idToken: null,
      refreshToken: null,
    });
  });
});