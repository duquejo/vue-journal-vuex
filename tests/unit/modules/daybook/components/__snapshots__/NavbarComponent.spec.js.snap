// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent.vue should show the component as needed 1`] = `
<nav class="navbar bg-primary"><a href="" class="navbar-brand text-white"><img src="assets/logo.png" alt="Vue logo" class="inline-block align-text-top mx-2" height="24"> John Doe</a>
  <div class="d-flex"><button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
